### ISSUE TITLE
Video Analysis 

### ISSUE THREAD

**Issue by sesathishkumar** [sesathishkumar/sskengineering#1]
import cv2
import torch
import torchvision
import torchvision.transforms as T
import numpy as np

# ✅ Load pretrained video action recognition model
# (TorchVision provides pretrained models like 'r3d_18')
model = torchvision.models.video.r3d_18(pretrained=True)
model.eval()

# Action classes for Kinetics-400 dataset
KINETICS_CLASSES = torchvision.datasets.Kinetics400.classes

# Video transforms
transform = T.Compose([
    T.Resize((112, 112)),
    T.ToTensor(),
    T.Normalize((0.43216, 0.394666, 0.37645),
                (0.22803, 0.22145, 0.216989)),
])

# Open video stream (0 = webcam, or replace with "video.mp4")
cap = cv2.VideoCapture(0)

# Buffer to hold frames for action recognition
clip = []

while cap.isOpened():
    ret, frame = cap.read()
    if not ret:
        break

    # Preprocess frame
    img = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
    img = cv2.resize(img, (128, 128))
    img_tensor = transform(img)
    clip.append(img_tensor)

    # Use 16-frame clip for prediction
    if len(clip) == 16:
        inputs = torch.stack(clip).unsqueeze(0)  # Shape: (1, 16, 3, 112, 112)
        inputs = inputs.permute(0, 2, 1, 3, 4)   # (N, C, T, H, W)

        with torch.no_grad():
            outputs = model(inputs)
            probs = torch.nn.Softmax(dim=1)(outputs)
            top5 = torch.topk(probs, k=5)

        # Get top predicted action
        pred_class = KINETICS_CLASSES[top5.indices[0][0]]
        pred_prob = top5.values[0][0].item()

        # Display on frame
        cv2.putText(frame, f"Action: {pred_class} ({pred_prob:.2f})",
                    (10, 30), cv2.FONT_HERSHEY_SIMPLEX, 1,
                    (0, 0, 255) if pred_prob > 0.5 else (0, 255, 0), 2)

        # Reset clip
        clip = []

    # Show video feed
    cv2.imshow("Suspicious Action Recognition", frame)

    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

cap.release()
cv2.destroyAllWindows()
